---
title: "piRNAs preparatory work"
output: rmarkdown::github_document
author: AlmorÃ² Scarpa
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(tidyverse)
```

Remove the adapter sequence and put then new files in a folder
```{bash, eval=FALSE}
for file in *.fastq.gz; do filename=$(basename "$file" .fastq); cutadapt -j 10 -a TGGAATTCTCGGGTGCCAAGG -o "./noadapt/$filename" "$file"; done
```

In parallel remove the extra column and name the new files .clean:
```{bash, eval=FALSE}
ls *.fastq.gz | parallel 'output_file=$(basename {} .fastq.gz); gzip -cd {} | awk "{print \$1}" | gzip -c > "$output_file.clean.fastq.gz"'
```

Size filter, this will keep only small RNAs:
```{bash, eval=FALSE}
ls *.clean.fastq.gz | parallel 'output_file=$(basename {} .clean.fastq.gz); gzip -cd {} | paste - - - - | awk "length(\$2) > 17 && length(\$2) < 36" | tr "\t" "\n" | gzip -c > "$output_file.trim.fastq.gz"'
```

Novoalign to index the fast file with all the RNAs:
```{bash, eval=FALSE}
novoindex newfilename.nvi fastafile
```
The fasta files with all the transcript, non-coding RNAs and miRNAs can be dowloaded from Flybase
But they need to be modified changing the header, keeping the ID and adding "_type", type can be either mRNA, miRNA or ncRNA.
Then cat with a te library, TEs header must end with "_te"

Map wit Novoalign:
```{bash, eval=FALSE}
for file in *.trim.fastq.gz; do filename=$(basename "$file" .trim.fastq.gz); gzip -cd "$file" | /Applications/novocraft/novoalign -d /path/to/folder/TEs_and_RNAs.nvi -f - -F STDFQ -o SAM -o FullNW -r RANDOM > "${filename}.sam"; done
```

Make directory bam:
```{bash, eval=FALSE}
mkdir bam
```

Turn .sam into bam:
```{bash, eval=FALSE}
for i in *.sam ; do n=${i%.sam} ; samtools view -Sb $i > bam/$n.bam ; done
```

Sort the .bam file:
```{bash, eval=FALSE}
for i in *.bam ; do n=${i%.bam} ; samtools sort $i -o $n.sort ; done
```

piRNAs on TEs:
```{bash, eval=FALSE}
for i in *.sort; do n=${i%.sort}; samtools view $i |python /path/to/folder/piRNA-distribution-onTE.py --sam - --sample-id $n > $n-graph-on-TE.forR ; done
```

Ping-pong:
```{bash, eval=FALSE}
for i in *.sort; do n=${i%.sort}; samtools view $i | python /path/to/folder/ping-pong-signature.py --sam - --max-mm 2 --sample-id $n > $n.pps ; done
```
