---
title: "Drosophila sechellia - PCA"
output: rmarkdown::github_document
editor_options: 
  markdown: 
    wrap: sentence
    author: Riccardo
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(knitr))
suppressPackageStartupMessages(library(kableExtra))
suppressPackageStartupMessages(library(ggpubr))
suppressPackageStartupMessages(library(maps))
suppressPackageStartupMessages(library(RgoogleMaps))
theme_set(theme_bw())
```

```{r}
(metadata <- read_tsv("/Volumes/EXT-RICCARDO/DoubleTrouble/Dsec/metadata.txt", show_col_types = FALSE))

pca_metadata <- metadata %>% select(run_accession, island, year)

PCs <- c("PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7", "PC8", "PC9", "PC10", "PC11", "PC12", "PC13", "PC14", "PC15", "PC16", "PC17", "PC18", "PC19", "PC20")
```

## PCA

To prepare the data for the PCA, I first downloaded, trimmed and merged (_1 and _2) all the *D. sechellia* strains present in our dataset. I also downloaded the reference genome of *D. sechellia* from the NCBI database (GCF_004382195.2_ASM438219v2_genomic.fna).

I mapped the short reads of each sample to the **reference genome** and then sorted the **bam** files.
```
find . -name "*.fastq.gz" | parallel -j 4 'n={/.}; bwa bwasw -t 4 /Volumes/Temp/Riccardo/simulans-clade-TE-invasions/sechellia-population-structure/GCF_004382195.2_ASM438219v2_genomic.fna {} | samtools sort -@ 4 -m 3G - > /Volumes/Temp/Riccardo/simulans-clade-TE-invasions/sechellia-population-structure/${n}.sort.bam'
```

I generated the mpileup (**bcf**) file for all the bam files.
```
find . -name "*.bam" | parallel -j 20 'n={/.}; samtools mpileup -uf /Volumes/Temp/Riccardo/simulans-clade-TE-invasions/sechellia-population-structure/reference/GCF_004382195.2_ASM438219v2_genomic.fna {} -o ${n}.bcf'
```

I converted the **bcf** into **vcf**, performing the variants call using **bcftools**. Then I zipped all the files using **bgzip** and **indexed** the bcf files with bcftools.
```
find . -name "*.bcf" | parallel -j 20 'n={/.}; bcftools call -mv -Ov {} -o ${n}.vcf'
find . -name "*.vcf" | parallel -j 20 'n={/.}; bgzip {}'
find . -name "*.vcf.gz" | parallel -j 20 'n={/.}; bcftools index {}'
```

After, I merged the vcf files.
```
bcftools merge *vcf.gz -o dsec.vcf
```

Using **plink2**, I pruned the linked variants (see this tutorial for explanation <https://speciationgenomics.github.io/pca/>)
```
plink2 --vcf dsec.vcf --double-id --allow-extra-chr  --indep-pairwise 50 10 0.1 --out dsec.pruned
```

And finally, performed the PCA.
```
plink2 --vcf dsec.vcf --double-id --allow-extra-chr --extract dsec.pruned.prune.in --make-bed --pca --out dsec.pca
```

The PCA gives as output files the **.eigenvec** file (containing the values of each PC for each sample) and the **.eigenval** file (variability explained by each PC, easily convertible in % value).

This function takes as input the two output files (vec, val) and the metadata. Note that depending on your metadata column names you might need to change the code below when plotting or merging.
Also, the names of the samples in the eigenvec files must be adjusted (es. removing part of the file path) to match with the metadata and make the merging possible. This is done here in the function by removing everything before "SRR" and everything after the last number after "SRR". Might need adjustments.
```{r}
eigen2pca <- function(vec, val, metadata) {
  eigenvec <- read_table(vec, col_names = c("run_accession", "run_accession2", "PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7", "PC8", "PC9", "PC10", "PC11", "PC12", "PC13", "PC14", "PC15", "PC16", "PC17", "PC18", "PC19", "PC20")) %>% select(-run_accession2) %>% mutate(run_accession = str_extract(run_accession, "SRR\\d+"))

eigenval <- read_table(val, col_names = c("val")) %>% mutate(PC = PCs, variability_explained = paste0(round((val/sum(val)*100),2), "%"))

var_explained <- eigenval %>% select(variability_explained) %>% pull()
 
pcaable <- inner_join(pca_metadata, eigenvec, by="run_accession")

(pca_plot1 <- ggplot(pcaable, aes(x=PC1, y=PC2, color=island)) + geom_point(alpha=0.7) +
    xlab(paste0("PC1: ", var_explained[1])) + ylab(paste0("PC2: ", var_explained[2])))
}
```

The first PCA is performed using all the strains of Sechellia.
The second one is performed removing the 2 suspicious strains of unknown origin.
```{r}
eigen2pca("/Volumes/EXT-RICCARDO/DoubleTrouble/Dsec/population-structure/v1/dsec.pca.eigenvec", "/Volumes/EXT-RICCARDO/DoubleTrouble/Dsec/population-structure/v1/dsec.pca.eigenval", pca_metadata)

#eigen2pca("/Volumes/EXT-RICCARDO/DoubleTrouble/Dsec/population-structure/v2/dsec.pca.eigenvec", "/Volumes/EXT-RICCARDO/DoubleTrouble/Dsec/population-structure/v2/dsec.pca.eigenval", pca_metadata)

eigen2pca("/Volumes/EXT-RICCARDO/DoubleTrouble/Dsec/population-structure/v3/dsec.pca.eigenvec", "/Volumes/EXT-RICCARDO/DoubleTrouble/Dsec/population-structure/v3/dsec.pca.eigenval", pca_metadata)
```